v2/pkg/protocols/websocket/websocket.go

```diff
@@ -0,0 +1,380 @@
+	payloadValues["Path"] = parsed.Path
+
+	for key, value := range request.Headers {
+		finalData, dataErr := expressions.EvaluateByte([]byte(value), payloadValues)
+		if dataErr != nil {
+			request.options.Output.Request(request.options.TemplateID, input, "websocket", dataErr)
+			request.options.Progress.IncrementFailedRequestsBy(1)
+			return errors.Wrap(dataErr, "could not evaluate template expressions")
+		}
+		header.Set(key, string(finalData))
+	}
+	websocketDialer := ws.Dialer{
+		Header:    ws.HandshakeHeaderHTTP(header),
+		Timeout:   time.Duration(request.options.Options.Timeout) * time.Second,
+		NetDial:   request.dialer.Dial,
+		TLSConfig: &tls.Config{InsecureSkipVerify: true, ServerName: hostname},
+	}
+
+	finalAddress, dataErr := expressions.EvaluateByte([]byte(request.Address), payloadValues)
+	if dataErr != nil {
+		request.options.Output.Request(request.options.TemplateID, input, "websocket", dataErr)
+		request.options.Progress.IncrementFailedRequestsBy(1)
+		return errors.Wrap(dataErr, "could not evaluate template expressions")
+	}
+
+	addressToDial := string(finalAddress)
+	parsedAddress, err := url.Parse(addressToDial)
+	if err != nil {
+		request.options.Output.Request(request.options.TemplateID, input, "websocket", dataErr)
+		request.options.Progress.IncrementFailedRequestsBy(1)
+		return errors.Wrap(dataErr, "could not parse input url")
+	}
+
+	if parsed.Path != "" && parsed.Path != "/" {
+		parsedAddress.Path = path.Join(parsedAddress.Path, parsed.Path)
+		addressToDial = parsedAddress.String()
+	}
+
+	conn, readBuffer, _, err := websocketDialer.Dial(context.Background(), addressToDial)
+	if err != nil {
+		request.options.Output.Request(request.options.TemplateID, input, "websocket", err)
+		request.options.Progress.IncrementFailedRequestsBy(1)
+		return errors.Wrap(err, "could not connect to server")
+	}
+	defer conn.Close()
+
+	responseBuilder := &strings.Builder{}
+	if readBuffer != nil {
+		_, _ = io.Copy(responseBuilder, readBuffer) // Copy initial response
+	}
+
+	events, requestOutput, err := request.readWriteInputWebsocket(conn, payloadValues, input, responseBuilder)
+	if err != nil {
+		request.options.Output.Request(request.options.TemplateID, input, "websocket", err)
+		request.options.Progress.IncrementFailedRequestsBy(1)
+		return errors.Wrap(err, "could not read write response")
+	}
+	request.options.Progress.IncrementRequests()
+
+	if request.options.Options.Debug || request.options.Options.DebugRequests {
+		gologger.Debug().Str("address", input).Msgf("[%s] Dumped Websocket request for %s", request.options.TemplateID, input)
+		gologger.Print().Msgf("%s", requestOutput)
+	}
+
+	request.options.Output.Request(request.options.TemplateID, input, "websocket", err)
+	gologger.Verbose().Msgf("Sent Websocket request to %s", input)
+
+	data := make(map[string]interface{})
+	for k, v := range previous {
+		data[k] = v
+	}
+	for k, v := range events {
+		data[k] = v
+	}
+	data["success"] = "true"
```

Comment:
same here

Reply:
https://github.com/projectdiscovery/nuclei/issues/1218 - will be handled here